### PR TITLE
Subscriptions filtering

### PR DESCRIPTION
**Pull request by a-alle**
# Description

Implements types as per the new API, subscriptions filtering and relationshipDeleted events on the disconnect operation

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at
 https://neo4j-graphql-toolbox-stage-2383.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/src/schema/resolvers/subscriptions/where.ts [hunk 1]
```diff
@@ -36,7 +50,17 @@ export function subscriptionWhere(
         return filterByProperties(node, where, event.properties.old);
     }
     if (event.event === "connect") {
-        return filterRelationshipConnectionsByProperties(node, where, event);
+        if (!nodes || !relationshipFields) {
+            // throw?
```
Comment (angrykoala) [inline packages/graphql/src/schema/resolvers/subscriptions/where.ts:54]: ? :smile:

File: packages/graphql/src/schema/subscriptions/generate-subscription-types.ts [hunk 2]
```diff
@@ -17,8 +17,6 @@
  * limitations under the License.
  */
 
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-nocheck
```
Comment (angrykoala) [inline packages/graphql/src/schema/subscriptions/generate-subscription-types.ts:21]: No idea what happened here, but great to remove it :laughing:

File: packages/graphql/tests/e2e/subscriptions/filtering/connect.e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,1934 @@
...
+                            ${typeMovie.plural} {
+                                title
+                            }
+                        }
+                    }
+                `,
+            })
+            .expect(200);
+        await delay(3);
+        console.log(wsClient.events);
```
Comment (a-alle) [inline packages/graphql/tests/e2e/subscriptions/filtering/connect.e2e.test.ts:710]: console.log

File: packages/graphql/src/schema/subscriptions/generate-subscription-types.ts [hunk 4]
```diff
@@ -453,28 +429,22 @@ function _buildRelationshipFieldDestinationTypes({
...
+        return [
+            ...relationshipProperties.primitiveFields,
+            ...relationshipProperties.enumFields,
+            ...relationshipProperties.scalarFields,
+            ...relationshipProperties.temporalFields,
+            ...relationshipProperties.pointFields,
+        ];
     }
 }
 // TODO: move this + helpers to separate file
```
Comment (a-alle) [inline packages/graphql/src/schema/subscriptions/generate-subscription-types.ts:450]: extract to separate file

File: packages/graphql/src/schema/subscriptions/generate-subscription-types.ts [hunk 5]
```diff
@@ -328,7 +297,14 @@ function getRelationshipEventDataForNode(
     destinationProperties: Record<string, any>;
     destinationTypename: string;
 } {
-    if (event.toTypename === node.name) {
+    let condition = event.toTypename === node.name;
+    if (event.toTypename === event.fromTypename) {
+        // must check relationship direction from schema
+        // TODO: memoize result
```
Comment (a-alle) [inline packages/graphql/src/schema/subscriptions/generate-subscription-types.ts:303]: fix todo

File: packages/graphql/src/schema/subscriptions/generate-subscription-types.ts [hunk 6]
```diff
@@ -344,187 +315,24 @@ function getRelationshipEventDataForNode(
...
-        })
-        .reduce((acc, { relationshipFieldType, fieldName }) => {
-            if (relationshipFieldType && _hasProperties(relationshipFieldType)) {
-                acc[fieldName] = relationshipFieldType;
-            }
-            return acc;
-        }, {});
+    relationshipName: string;
+}): RelationField | undefined {
+    let relationshipNameToRelationField: Map<string, RelationField | undefined>;
```
Comment (a-alle) [inline packages/graphql/src/schema/subscriptions/generate-subscription-types.ts:326]: @angrykoala any opinions on this memoization attempt here?

**Comment by a-alle** [comment]
First code review changes done. Please revisit

**Pull request merged**